server/src/main/java/org/eclipse/openvsx/util/LogUtils.java

```diff
@@ -0,0 +1,827 @@
+ * other threads.
+ *
+ * <p>To use <strong>Asynchronous nested messages</strong>, see {@link #traceAsyncStart(Logger,
+ * Level, String, String, int, Object...)}, and {@link #traceAsyncEnd(Logger, Level, String, String,
+ * int, Object...)}
+ *
+ * <p>To use <strong>Object tracking</strong>, see {@link #traceObjectCreation(Logger, Level,
+ * Object)} and {@link #traceObjectDestruction(Logger, Level, Object)}
+ *
+ * <p>The design philosophy of this class is very heavily inspired by the trace event format of
+ * Google. The full specification is available <a
+ * href=https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/edit?pli=1#>here</a>.
+ *
+ * <p>The main goals are clarity of output and simplicity for the developer. Performance is a nice
+ * to have, but is not the main concern of this helper. A minor performance impact compared to
+ * simply logging the events is to be expected.
+ *
+ * @author Matthew Khouzam
+ * @noinstantiate This class is not intended to be instantiated by clients. It is a helper class.
+ */
+public final class LogUtils {
+
+  private static final Format FORMAT = new DecimalFormat("#.###"); // $NON-NLS-1$
+
+  /*
+   * Field names
+   */
+  private static final String ARGS = "args"; // $NON-NLS-1$
+  private static final String NAME = "name"; // $NON-NLS-1$
+  private static final String CATEGORY = "cat"; // $NON-NLS-1$
+  private static final String ID = "id"; // $NON-NLS-1$
+  private static final String TID = "tid"; // $NON-NLS-1$
+  private static final String PID = "pid"; // $NON-NLS-1$
+  private static final String TIMESTAMP = "ts"; // $NON-NLS-1$
+  private static final String PHASE = "ph"; // $NON-NLS-1$
+
+  private static final String ARGS_ERROR_MESSAGE =
+      "Data should be in the form of key, value, key1, value1, ... TraceCompassScopeLog was supplied "; //$NON-NLS-1$
+  private static final AtomicInteger ID_GENERATOR = new AtomicInteger(0);
+
+  private LogUtils() {
+    // do nothing
+  }
+
+  /**
+   * Scope Logger helper. This will automatically log entry and exit of the scope. This scope log
+   * will be shown under any scope enclosing it, but will not be the source, or destination of any
+   * link to other scopes. If relations should be done with other scopes, the {@link FlowScopeLog}
+   * class is more appropriate.
+   *
+   * <p>Usage:
+   *
+   * <pre>
+   * {@code usage of ScopeLog}
+   *  try (ScopeLog linksLogger = new ScopeLog(LOGGER, Level.CONFIG, "Perform Query")) { //$NON-NLS-1$
+   *      ss.updateAllReferences();
+   *      dataStore.addAll(ss.query(ts, trace));
+   *  }
+   * </pre>
+   *
+   * <p>will generate the following trace
+   *
+   * <pre>
+   * {@code trace output}
+   *  INFO: {"ts":12345,"ph":"B",tid:1,"name:Perform Query"}
+   *  INFO: {"ts":"12366,"ph":"E","tid":1}
+   * </pre>
+   */
+  public static class ScopeLog implements AutoCloseable {
+
+    private final long fTime;
```

Comment:
It looks like `f<name>` pattern is not used anywhere else in the code, and even the PR contains another style to name the class members. I would propose to choose a single style and then stick to it.